axelor-stock/src/main/java/com/axelor/apps/stock/db/repo/LogisticalFormStockRepository.java

```diff
@@ -34,19 +39,40 @@
   @Override
   public LogisticalForm save(LogisticalForm logisticalForm) {
     try {
-      if (Strings.isNullOrEmpty(logisticalForm.getDeliveryNumber())
-          && logisticalForm.getCompany() != null) {
-        String sequenceNumber =
-            Beans.get(SequenceService.class)
-                .getSequenceNumber("logisticalForm", logisticalForm.getCompany());
-        if (Strings.isNullOrEmpty(sequenceNumber)) {
-          throw new AxelorException(
-              Sequence.class,
-              TraceBackRepository.CATEGORY_NO_VALUE,
-              I18n.get(IExceptionMessage.LOGISTICAL_FORM_MISSING_SEQUENCE),
-              logisticalForm.getCompany().getName());
+      Company company = logisticalForm.getCompany();
+
+      if (company != null) {
+        if (Strings.isNullOrEmpty(logisticalForm.getDeliveryNumber())) {
+          String sequenceNumber =
+              Beans.get(SequenceService.class)
+                  .getSequenceNumber("logisticalForm", logisticalForm.getCompany());
+          if (Strings.isNullOrEmpty(sequenceNumber)) {
+            throw new AxelorException(
+                Sequence.class,
```

Comment:
Please use select value instead of string literal.

Reply:
done